nucliadb_paragraphs3/src/writer.rs

```diff
@@ -231,10 +231,25 @@ impl ParagraphWriterService {
         let mut paragraph_counter = 0;
 
         for (field, text_info) in &resource.texts {
+            let chars: Vec<char> = REGEX.replace_all(&text_info.text, " ").chars().collect();
+            let resource_facets = resource
+                .labels
+                .iter()
+                .map(Facet::from_text)
+                .collect::<Result<Vec<_>, _>>()
+                .map_err(|e| tantivy::TantivyError::InvalidArgument(e.to_string()))?;
+            let field_facets = text_info
+                .labels
+                .iter()
+                .map(Facet::from_text)
+                .collect::<Result<Vec<_>, _>>()
+                .map_err(|e| tantivy::TantivyError::InvalidArgument(e.to_string()))?;
+            let field_labels: Vec<_> =
```

Comment:
Do you need to collect twice? Isn't it better to postpone it the second time to the `chain` of `paragraph_labels`? You can find that approach at: https://github.com/nuclia/nucliadb/blob/ea225f385dbc39f97a266056b071edfdbdfff102/nucliadb_paragraphs2/src/writer.rs#L253